etc/node_ws/app.js

```diff
@@ -0,0 +1,105 @@
+const package = require('./package.json');
+const stan = require('node-nats-streaming');
+const WebSocket = require('ws');
+const {uuid} = require('uuidv4');
+
+class Handler {
+  constructor(config, socket, request) {
+    this.config = config;
+    this.id = uuid();
+    this.ip = request.connection.remoteAddress;
+    this.socket = socket;
+
+    let seq = this.getSequence(request.url);
+    if (seq == null) {
+      console.log('Sequence not provided by ' + this.ip + ' or is not an integer; closing connection.');
+      this.socket.close(1008, 'Malformed/nonexistent sequence');
+    }
+    this.sequence = seq;
+  }
+  start() {
+    console.log('Connection opened from IP: ' + this.ip);
+
+    // connect to nats streaming
+    console.log('Connecting to stan server for client ' + this.ip + ' with id ' + this.id);
+    let connection = stan.connect(this.config.cluster_id, this.id);
+    connection.on('connect', (connection) => this.onStanConnection(connection));
+
+    // define socket behavior
+    this.socket.isAlive = true;
+    this.socket.on('close', () => {
+      connection.close();
+      console.log('Connection closed for client ' + this.ip);
+    });
+    this.socket.on('pong', () => {
```

Comment:
Ping/pong and isAlive should all be in the App, and not the Handler.

Reply:
Moved ping behavior to App class